File: fixtures/scale/java/large/src/cryptobench/ciphers/xorshift/XorShiftKey.java
package cryptobench.ciphers.xorshift;

public class XorShiftKey {
    private final long seed;

    public XorShiftKey(long seed) {
        this.seed = seed;
    }

    public long getSeed() {
        return seed;
    }

    public static XorShiftKey defaultKey() {
        return new XorShiftKey(0x1A2B3C4D5E6FL);
    }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/xorshift/XorShiftKey.scala
package cryptobench.ciphers.xorshift

/** Parameters for the xorshift cipher. */
final case class XorShiftKey(seed: Long) {
  def isZeroSeed: Boolean = seed == 0L
}

object XorShiftKey {
  def default(): XorShiftKey = XorShiftKey(0x1A2B3C4D5E6FL)
}
